# Copyright 2023 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

syntax = "proto3";

message CatalogRequest {
    int32 max_results = 1;
    string product_id = 2;
}

message CatalogItem {
    string id = 1;
    string title = 2;
    string author = 3;
    string description = 4;
    int32 rating = 5;
    string price_dollars = 6;
    string price_cents = 7;
}

message CatalogResponse {
    repeated CatalogItem items = 1;
}

service Catalog {
    rpc Catalog (CatalogRequest) returns (CatalogResponse);
}
